core/app/models/spree/adjustment.rb

```diff
@@ -56,6 +56,17 @@ class Adjustment < Spree::Base
     extend DisplayMoney
     money_methods :amount
 
+    # Returns Adjustments of completed Orders.
+    #
+    # @param excluded_orders [Array<Spree::Order>] Orders to exclude from query
+    # @return [ActiveRecord::Relation] Scoped Adjustments
+    def self.in_completed_orders(excluded_orders: [])
+      joins(:order).
+      merge(Spree::Order.complete).
+      where.not(spree_orders: { id: excluded_orders }).
+      distinct
```

Comment:
Not strictly related to this PR, since the `distinct` comes from the previous implementation. Just a heads-up.

Last time I checked it, the generated query was very slow because of `DISTINCT(*)` on a relatively big table (~400k rows). It was something like ~1ms query response with `group by` vs ~500ms using `distinct(*)` on my local machine.
It's worth double checking it and, if confirmed, opening a separate issue.

Reply:
This could be the right time to speed up the query, I'll profile both versions with some test data, thanks.

Reply:
I tested it on about 500k rows of random genereated rows but I couldn't find any big difference. If you have a more realistic dump without sensitive data could you please share it with me?

Reply:
🤔 thanks for looking into this. I'll try to create a reproducible case with dummy data then eventually will open a separate issue with steps to reproduce it.